pkg/patterns/addon/pkg/status/version.go

```diff
@@ -0,0 +1,81 @@
+package status
+
+import (
+	"context"
+	"fmt"
+	"github.com/blang/semver"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	addonsv1alpha1 "sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/apis/v1alpha1"
+
+	//"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative/pkg/manifest"
+)
+
+// NewVersionCheck provides an implementation of declarative.Reconciled that
```

Comment:
Oh cool - using a signature inspired by Reconcilers is an interesting pattern to explore!